open_schools_platform/photo_management/photos/services.py

```diff
@@ -0,0 +1,28 @@
+from rest_framework.exceptions import PermissionDenied
+
+from open_schools_platform.common.services import model_update
+from open_schools_platform.common.utils import filter_dict_from_none_values
+from open_schools_platform.photo_management.photos.models import Photo
+from open_schools_platform.user_management.users.models import User
+
+
+def create_photo(image: bytes = None) -> Photo:
+    photo = Photo.objects.create_photo(
+        image=image
+    )
+    return photo
+
+
+def update_photo(*, photo: Photo, data, user: User = None) -> Photo:
+    non_side_effect_fields = ['image']
+    filtered_data = filter_dict_from_none_values(data)
+
+    if user and photo and not user.has_perm('photos.photo_access', photo):
+        raise PermissionDenied
```

Comment:
no it works in other way
permission check locates in selector layer, move this logic to get_photo